Automattic/wp-calypso
# Move add products memberships endpoint to v2

Related to 106-gh-gold
Blocked by https://github.com/Automattic/jetpack/pull/32412
Blocks D118667-code

## Proposed Changes

* Move calls to add products endpoint to v2 for memberships (after https://github.com/Automattic/jetpack/pull/32412 is deployed)

## Testing Instructions



1. If you are an a11n, you can patch D118667-code and sandbox public API.
2. If you'd like to test the entire flow, you can use the release branch in Jetpack Beta on your Jetpack site. In the upper-right corner of the wp-admin interface of your Jetpack site, click Jetpack Beta, then the first option, then "Manage". Paste the following into the Release branch option and click "Activate" when the result is found for: refactor/add-products-v1-to-v2-proxied-endpoint.
3. If you're an a11n, you can download the patched version of Jetpack to your sandbox with `bin/jetpack-downloader test jetpack refactor/add-products-v1-to-v2-proxied-endpoint`.
4. On your Jetpack site, connect Jetpack to your WPCOM account, and add a new post with the "Subscribe" payment block.
5. Follow the prompts to setup a payment plan. Making note of if and when you are transferred to wordpress.com. If you see this, do not continue on wordpress.com - instead, continue to the next step, where we will use the patched Calypso to create a new payment plan.
6. Open calypso via the Calypso live links on this PR.
7. Once calypso loads, append `/earn/payment-plans` to the end of the url and hit enter. If you get "You are not authorized to view this page" click "switch site" and select your Jetpack site.
8. You should now be at the calypso.live version of `/earn/payment-plans/<your jetpack domain>`. If this is a new Jetpack site, you'll be greeted with an empty list of payment plans.
9. Click to add a new payment plan. The plan should add successfully.
10. Click to edit the created payment plan, make some adjustments and click save. The plan should save successfully and your edits should persist if you refresh the page.
11. Click to delete the created payment plan. The plan should delete, and if you refresh the page the plan should not re-appear in the list of payment plans.
12. Return to your Jetpack site and create a new post, this time with a payment button. Click the icon to select the payment plan and you should see a list of payment plans. Click add new payment plan and fill out the form, clicking save when done. The plan should exist when you return to the calypso live page and view your payment plans.

## Pre-merge Checklist
